internal/store/cluster.go

```diff
@@ -0,0 +1,164 @@
+package store
+
+import (
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+// Cluster represents a Kubernetes cluster as tracked by the store.
+type Cluster struct {
+	ID                  string
+	Provider            string
+	Provisioner         string
+	ProviderMetadata    []byte
+	ProvisionerMetadata []byte
+	AllowInstallations  bool
+	CreateAt            time.Time
+	DeleteAt            *time.Time
+	LockAcquiredBy      *string
+	LockAcquiredAt      *time.Time
+}
+
+// Clone returns a deep copy the cluster.
+func (c *Cluster) Clone() *Cluster {
+	var clone Cluster
+	data, _ := json.Marshal(c)
```

Comment:
Any risks not checking/handling the error?

Reply:
My gut says no -- any marshalling issue is likely to apply to all clusters and not a particular cluster. 0/5 on continuing to omit, though.